syntax = "proto3";

package rds;

import "google/protobuf/empty.proto";
import "common.proto";
import "proxy.proto";

service JobService {
  rpc Listen(google.protobuf.Empty) returns (stream JobResponse);
  rpc GetStatus(JobIdRequest) returns (JobStatusResponse);
  rpc GetJob(JobIdRequest) returns (JobResponse);
  rpc Schedule(JobIdRequest) returns (JobStatusResponse);
  rpc Cancel(JobIdRequest) returns (JobStatusResponse);
  rpc Acknowledge(JobIdRequest) returns (google.protobuf.Empty);
}

message JobIdRequest {
  uint64 id = 1;
}

message JobStatusResponse {
  Status status = 1;
}

message JobResponse {
  oneof kind {
    ProxyCreateRequest proxy_create_request = 1;
    ProxyInfoResponse proxy_info_response = 2;
    ProxyIdRequest proxy_id_request = 3;
    ProxyStatusResponse proxy_status_response = 4;
    ProxyMonitorCredentialResponse proxy_monitor_credential_response = 5;
    ProxyUpgradeRequest proxy_upgrade_request = 6;
    ProxyDeleteResponse proxy_delete_response = 7;
  }
}
